packages/debug/src/browser/view/debug-watch-expression.tsx

```diff
@@ -37,16 +38,19 @@ export class DebugWatchExpression extends ExpressionItem {
         await super.evaluate('watch');
     }
 
-    protected override setResult(body?: DebugProtocol.EvaluateResponse['body']): void {
-        // overridden to ignore error
-        super.setResult(body);
+    protected override setResult(body?: DebugProtocol.EvaluateResponse['body'], error?: string): void {
+        if (!this.options.session()) {
+            return;
+        }
```

Comment:
I've noticed that vscode yields this (generic but fitting) result when no session is active:

![image](https://user-images.githubusercontent.com/4377073/205678820-6562014f-73d3-4c80-8073-831cbaa5f918.png)

Does that make sense for us as well? Or is that out of scope?

Reply:
@msujew I believe we do too with these changes? I did notice that we do not update `watch expressions` properly which I wanted to file a bug for, basically if a new debug session is execute the existing `watch expressions` are not updated.

Reply:
Right, I think it's the overall issue on master where our watch expressions do not seem to be properly updated.

For example:

![image](https://user-images.githubusercontent.com/40359487/205685181-bdfca8d9-d8d4-434b-a1aa-47ae76a67d26.png)

If I do not have an active debug session and set the `value` watch expression, hitting the breakpoint when I do start a session does not update the expression like it does in vscode.

![image](https://user-images.githubusercontent.com/40359487/205685390-94db670a-8e3f-4eae-beff-9b4ce9c3f38f.png)

Reply:
I see, alright. Then it's out of scope 👍